### PR TITLE
sql: crdb_internal.fingerprint isn't Immutable

### PR DESCRIPTION
**Pull request by stevendanna**
Epic: none

Release note: None

### CODE REVIEW

**Review by msbutler — APPROVED** [review]

File: pkg/sql/sem/builtins/builtins.go [hunk 1]
```diff
@@ -7660,7 +7660,7 @@ expires until the statement bundle is collected`,
 				}
 			},
 			Info:       "This function is used only by CockroachDB's developers for testing purposes.",
-			Volatility: volatility.Immutable,
+			Volatility: volatility.Stable,
```
Comment (dt) [inline pkg/sql/sem/builtins/builtins.go:7663]: isn't it Volatile? it modifies tscache doesn't it ?

Comment (stevendanna) [reply pkg/sql/sem/builtins/builtins.go:7663]: I'm not sure I know what "counts" in terms of "the database" or "transaction state" here.  

> 	// Stable means that the operator cannot modify the database or the
	// transaction state and is guaranteed to return the same results given the
	// same arguments whenever it is evaluated within the same statement. Multiple
	// calls to a stable operator can be optimized to a single call.
	// Examples: current_timestamp, current_date.

We've made crdb_internal.scan() Stable as well under the assumption that it should return the same result in the same transaction and doesn't write data. Perhaps both need to change.

Comment (dt) [reply pkg/sql/sem/builtins/builtins.go:7663]: I guess there's no observable side effect of it running once vs twice.

**Review by dt — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=dt,msbutler

**Pull request merged**